dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageFactoryImpl.java

```diff
@@ -475,63 +486,52 @@ private void saveLanguageKeys(String fileLangName, Map<String, String> keys, Set
 			if (tempFile.exists())
 				tempFile.delete();
 			if (tempFile.createNewFile()) {
-				fileReader = new FileInputStream(file);
+				fileReader = Files.newInputStream(file.toPath());
 
 				tempFileWriter = new PrintWriter(tempFilePath, "UTF8");
 
 				for (String k : toDeleteKeys) {
 					keys.remove(k);
 				}
-				
-			
+
 				for(Map.Entry<String,String> newKey : keys.entrySet()) {
 					tempFileWriter.println(newKey.getKey() + "=" + newKey.getValue());
 				}
 			} else {
 				Logger.warn(this, "Error creating properties temp file: '" + tempFilePath + "' already exists.");
 			}
-		} catch (FileNotFoundException e) {
-			throw e;
 		} catch (IOException e) {
 			throw e;
 		} finally {
 			if(fileReader != null)
 				fileReader.close();
-			if(fileReader != null) {
+			if(tempFileWriter != null) {
 				tempFileWriter.flush();
 				tempFileWriter.close();
 			}
 
 		}
 		
-		FileChannel fileToChannel = null;
-		FileChannel fileFromChannel = null;
+		try(InputStream tempFileInputStream = Files.newInputStream(tempFile.toPath());
+			OutputStream fileOutputStream = Files.newOutputStream(file.toPath())) {
 
-		try {
 			if (file.exists() && tempFile.exists()) {
-				fileToChannel = (new FileOutputStream(file)).getChannel();
-				fileFromChannel = (new FileInputStream(tempFile)).getChannel();
-				fileFromChannel.transferTo(0, fileFromChannel.size(), fileToChannel);
+                final ReadableByteChannel inputChannel = Channels.newChannel(tempFileInputStream);
+                final WritableByteChannel outputChannel = Channels.newChannel(fileOutputStream);
+                FileUtil.fastCopyUsingNio(inputChannel, outputChannel);
+                inputChannel.close();
+                outputChannel.close();
 			} else {
 				if (!file.exists())
 					Logger.warn(this, "Error: properties file: '" + filePath + "' doesn't exists.");
 				if (!tempFile.exists())
 					Logger.warn(this, "Error: properties file: '" + tempFilePath + "' doesn't exists.");
 			}
-		} catch (FileNotFoundException e) {
-			throw e;
 		} catch (IOException e) {
 			throw e;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add logic to this catch clause or eliminate it and rethrow the exception automatically. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2737)